server/api/authentication/auth_handler.go

```diff
@@ -0,0 +1,79 @@
+package authentication
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/tobocop/go-teleport-directory-browser/api/session"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type AuthRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func (s *Server) AuthHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	var authReq AuthRequest
+	err := json.NewDecoder(req.Body).Decode(&authReq)
+	if err != nil {
+		log.Printf("AuthHandler json decode error: %v", err)
+		w.WriteHeader(400)
+	}
+
+	// TODO: Not sure if query escape is best option. Maybe base64?
+	escapedUser, err := url.QueryUnescape(authReq.Username)
```

Comment:
Why do username/password need to be escaped if they're sent in the request body?

Reply:
In theory, someone could enter a string for the username like: some-user","an-extra-field":"extra-value" and inject extra json fields into the payload. AuthReq doesn't have any other fields right now so there's nothing I can see that's obviously harmful, but felt better to add the extra security on a login form.

Reply:
It sure would! I got up in my head about writing a secure api and forgot that `JSON.stringify()` would sort it.